src/Analyzer.Utilities/Extensions/ISymbolExtensions.cs

```diff
@@ -147,6 +148,69 @@ public static bool MatchFieldByName(this ISymbol member, INamedTypeSymbol type,
         {
             return member != null && member.Kind == SymbolKind.Field && member.MatchMemberByName(type, name);
         }
+
+        /// <summary>
+        /// Checks if a given symbol implements an interface member implicitly or explicitly
+        /// </summary>
+        public static bool IsInterfaceMemberImplementation(this ISymbol symbol)
+        {
+            return symbol.IsExplicitInterfaceMemberImplementation() || symbol.IsImplicitInterfaceMemberImplementation();
+        }
+
+        /// <summary>
+        /// Checks if a given symbol implements an interface member implicitly
+        /// </summary>
+
+        public static bool IsImplicitInterfaceMemberImplementation(this ISymbol symbol)
```

Comment:
Can we make this method generic, with a constraint `where TSymbol: ISymbol`, and then modify the below to call `interfaceSymbol.GetMembers().OfType<TSymbol>()`? This way, clients which are passing in a specific symbol kind (say methods) are more performant.